src/components/enterprise-user-subsidy/UserSubsidy.jsx

```diff
@@ -16,28 +15,27 @@ export const UserSubsidyContext = createContext();
 
 const UserSubsidy = ({ children }) => {
   const { enterpriseConfig } = useContext(AppContext);
-  const [
-    subscriptionPlan,
-    isLoadingSubscriptionPlan,
-  ] = useEnterpriseCustomerSubscriptionPlan(enterpriseConfig.uuid);
   const [subscriptionLicense, isLoadingLicense] = useSubscriptionLicenseForUser(enterpriseConfig.uuid);
   const [offers, isLoadingOffers] = useOffers(enterpriseConfig.uuid);
 
   const isLoadingSubsidies = useMemo(
     () => {
-      const loadingStates = [isLoadingSubscriptionPlan, isLoadingLicense, isLoadingOffers];
+      const loadingStates = [isLoadingLicense, isLoadingOffers];
       return loadingStates.includes(true);
     },
-    [isLoadingSubscriptionPlan, isLoadingLicense, isLoadingOffers],
+    [isLoadingLicense, isLoadingOffers],
   );
 
+  // We return subscription plan as part of the license object now. This is to avoid renaming everything.
+  let subscriptionPlan = null;
   const contextValue = useMemo(
     () => {
       if (isLoadingSubsidies) {
         return {};
       }
       let hasAccessToPortal = false;
-      if (subscriptionPlan) {
+      if (subscriptionLicense) {
+        subscriptionPlan = subscriptionLicense.subscriptionPlan;
```

Comment:
Does it make sense to set `subscriptionPlan` inside of a `useMemo`, especially when that `useMemo` will re-compute `contextValue` any time the `subscriptionPlan` value changes?